### DISCUSSION TITLE
Enabling HTTPS for Applications Installed via Proxmox Script

### DISCUSSION THREAD

**Discussion by gamersalpha** [tteck/Proxmox#2852] [Help Needed]
Hello everyone,

I am using a Proxmox script from this GitHub repository: https://github.com/tteck/Proxmox, which facilitates the installation of applications using Proxmox containers. The script works well for deployment, but it installs all applications accessible only via HTTP. I would like to secure these applications by enabling HTTPS.

Could someone guide me on how to configure HTTPS for these containerized applications? Any advice on where to start, which certificates to use, and how to integrate them into the Proxmox environment would be greatly appreciated.

Thank you in advance for your help!

**Comment by di5cord20** [comment]
I use the Nginx Proxy Manager LXC script exactly for this purpose.  It allows you to use your own domain (which you can purchase from any regaistrar), and create access to all your LXCs accordingly.

**Comment by gamersalpha** [comment]
Thank you for the suggestion! While Nginx Proxy Manager LXC script is indeed a convenient way to manage access to containerized applications, it's important to note that transactions between the Nginx proxy and the machine are still transmitted in plaintext over the network. This means that sensitive information could potentially be intercepted by malicious actors.

For enhanced security, it's advisable to explore solutions that ensure end-to-end encryption for all network communications. This typically involves configuring HTTPS directly within the containerized applications themselves or implementing a secure reverse proxy setup with proper encryption and certificate management.

By prioritizing encryption at all layers of the network stack, we can better safeguard the confidentiality and integrity of our data, mitigating the risk of unauthorized access or interception.

Thank you for raising this important consideration!

**Reply by tteck** [reply]
If you suspect the presence of malicious actors within your private network, you have larger issues. 😉

**Reply by apitamy** [reply]
I'm old enough that i was around when firesheep was a thing. You could use it to take over peoples facebook sessions if you were connected to the same wifi because everything was http back then. I messed with wireshark and was able to see what people were searching for on google, again because everything was over http then. Some people live with roommates, have friends come over to get on wifi, or otherwise have insecure home networks. I think defaulting to https is not a ridiculous request. Just my 2 cents

**Comment by gamersalpha** [comment]
I appreciate your point, and you're absolutely correct that having malicious actors within a private network would indicate larger security concerns. However, even within seemingly secure environments, it's crucial to maintain a defense-in-depth strategy, which includes encrypting communications wherever possible.

Implementing secure practices, such as HTTPS, within local networks isn't just about defending against external threats; it's also about minimizing the risk posed by insider threats or potential breaches. Encryption helps protect data integrity and confidentiality, which are essential elements of any comprehensive security posture.

In response to your question about ensuring the integrity of your local network, I'm interested to hear if you have any particular practices or tools that you currently employ for network monitoring. Additionally, I'm open to any suggestions you might have for addressing this security concern within the local network.

**Comment by gamersalpha** [comment]
Hello,

i m still need to found a solution

how i can add a https directly on the virtual machine ?

**Reply by carsten-h** [reply]
> how i can add a https directly on the virtual machine ?

That differs from VM to VM. Sometimes the service also can use https on it's own, sometimes not. So there will be no general solution I think.

**Comment by carsten-h** [comment]
If you run all your VM and containers that you want to have https in one Proxmox server, you can create a virtual network without a real network connection where all these VM and containers will communicate. 
The only container which uses both this internal and the external net is the Nginx proxy manager.

So you can have the entire communication in your real net secure, only the virtual net where no one has connections into it will use http.
